File: backend/app/schemas/scheduler.py
from app.schemas.base import CamelModel


class SchedulerConfigRead(CamelModel):
    enabled: bool
    interval_minutes: int
    quiet_start: str
    quiet_end: str
    auto_confirm: bool
    last_run: str | None = None


class SchedulerConfigUpdate(CamelModel):
    enabled: bool | None = None
    interval_minutes: int | None = None
    quiet_start: str | None = None
    quiet_end: str | None = None
    auto_confirm: bool | None = None


class GenerationLogRead(CamelModel):
    id: str
    triggered_at: str
    trigger_reason: str
    events_proposed: int
    week_id: str | None = None
    success: bool
    error_message: str

File: frontend/src/api/scheduler.ts
import { apiGet, apiPatch } from './client'

export interface SchedulerConfig {
  enabled: boolean
  intervalMinutes: number
  quietStart: string
  quietEnd: string
  autoConfirm: boolean
  lastRun: string | null
}

export interface GenerationLog {
  id: string
  triggeredAt: string
  triggerReason: string
  eventsProposed: number
  weekId: string | null
  success: boolean
  errorMessage: string
}

export function getSchedulerConfig(): Promise<SchedulerConfig> {
  return apiGet('/api/scheduler/config')
}

export function updateSchedulerConfig(data: Partial<SchedulerConfig>): Promise<SchedulerConfig> {
  return apiPatch('/api/scheduler/config', data)
}

export function getGenerationLogs(limit?: number): Promise<GenerationLog[]> {
  const qs = limit ? `?limit=${limit}` : ''
  return apiGet(`/api/scheduler/logs${qs}`)
}
